### DISCUSSION TITLE
Can I pass positions (x,y) of nodes as arguments?

### DISCUSSION THREAD

**Discussion by shina181** [imohitmayank/jaal#56] [Q&A]
Hello Mohit,

Thanks for the code. 
I needed to know if node positions can be passed as arguments to the network.

Thanks and Regards,
Sheena :)

**Comment by imohitmayank** [comment]
Hey @shina181 yes it is possible. You need to do two things, 
- Fix the nodes by modifying the `vis_opts`. Example below
```python
# define vis options
vis_opts = {'height': '600px', # change height
            'interaction':{'hover': True}, # turn on-off the hover 
            'nodes': {'fixed': {'x': True, 'y': True}}}
```
- Add two columns in the `node_df` named `x` and `y` with their respective positions.  Example below, 
<img width="549" alt="image" src="https://user-images.githubusercontent.com/1883779/200304056-8b82ab26-5c90-4920-96ca-f0d0a48b8dfd.png">

Let me know incase of any issue 👋

**Reply by shina181** [reply]
Thanks Mohit ! I will try it out :)

**Comment by shina181** [comment]
Hi Mohit !
I tried what you suggested and it worked for the nodes. But for some reason now my edges are not visible in the final plot:
![image](https://user-images.githubusercontent.com/39149045/200837960-f5cc4992-c43c-4009-a33d-6f62b4949d53.png)

Here is a snippet of my code:
```
import pandas as pd
from jaal import Jaal

network_df=pd.read_csv('./network_df.csv', sep=',')
network_df.columns=['from', 'to', 'weight']
edge_df=network_df[network_df.weight!= 0.0000]
node_df=pd.read_csv('./nodes_3.csv', sep=',')
node_df.columns=['id', 'x', 'y']

app= Jaal(edge_df,node_df).plot(vis_opts={'height': '600px', # change height
                                      'edges': {'dashes':True, 'color': 'red',
                                                },
                                      'nodes': {'icon': {'size':20,'color':'#fadadd'}, 'shape':'icon',
                                                'fixed': {'x': True, 'y': True},
                                                'font':{'size':4},
                                                },
                                      'interaction':{'hover': False}, # turn on-off the hover
                                      'physics':{'stabilization':{'iterations': 2}},
                                        },
                                )

server = app.server
```
I also tried using edges.hidden: False. But still edges don't show up.

**Reply by imohitmayank** [reply]
Hey @shina181 it is working for me. 
<img width="623" alt="image" src="https://user-images.githubusercontent.com/1883779/201529211-3c6a3461-0cfd-479d-af98-05b2805306c1.png">

Here is my setting.
```
# define vis options
vis_opts = {'height': '600px', # change height
            'interaction':{'hover': True}, # turn on-off the hover 
            'nodes': {'fixed': {'x': True, 'y': True}}} 
```

Maybe you need to spread the `x` and `y` positions (I have mine between 1 and 1000).

**Reply by shina181** [reply]
Okay. Will try that and get back. Thanks :)
